cores/esp32/esp32-hal-rmt.c

```diff
@@ -616,446 +627,20 @@ rmt_obj_t* rmtInit(int pin, bool tx_not_rx, rmt_reserve_memsize_t memsize)
-//        RMT.tx_conf[channel].carrier_en = 0;
-//        RMT.tx_conf[channel].carrier_out_lv = 0;
-//        RMT.tx_conf[channel].tx_conti_mode = 0;
-//        RMT.tx_conf[channel].idle_out_lv = 0;     // signal level for idle
-//        RMT.tx_conf[channel].tx_start = 0;
-//        RMT.tx_conf[channel].tx_stop = 0;
-//        RMT.tx_conf[channel].carrier_eff_en = 0;
-//        RMT.tx_conf[channel].afifo_rst = 0;
-//        RMT.tx_conf[channel].conf_update = 0;
-//        RMT.tx_conf[channel].mem_tx_wrap_en = 0;
-//        RMT.tx_conf[channel].mem_rst = 1;
-
-        RMT.tx_conf[channel].idle_out_en = 1;     // enable idle
-        RMT.tx_conf[channel].div_cnt = 1;
-        RMT.tx_conf[channel].mem_size = buffers;
-        RMT.tx_conf[channel].mem_rd_rst = 1;
-        RMT.tx_conf[channel].conf_update = 1;
+        rmt_config_t config = RMT_DEFAULT_ARD_CONFIG_TX(pin, channel, buffers);
+        ESP_ERROR_CHECK(rmt_config(&config));
+        ESP_ERROR_CHECK(rmt_driver_install(config.channel, 0, 0));
+        log_d(" -- Init TX RMT on CH %d *RMT [%x]\n", channel, rmt);
     } else {
-        RMT.rx_conf[channel].conf0.div_cnt = 1;
-        RMT.rx_conf[channel].conf0.mem_size = buffers;
-        RMT.rx_conf[channel].conf0.carrier_en = 0;
-        RMT.rx_conf[channel].conf0.carrier_out_lv = 0;
-        RMT.rx_conf[channel].conf0.idle_thres = 0x80;
-        RMT.rx_conf[channel].conf1.rx_filter_en = 0;
-        RMT.rx_conf[channel].conf1.rx_filter_thres = 0;
-        RMT.rx_conf[channel].conf1.mem_rst = 0;
-        RMT.rx_conf[channel].conf1.mem_rx_wrap_en = 0;
-        RMT.rx_conf[channel].conf1.afifo_rst = 0;
-        RMT.rx_conf[channel].conf1.conf_update = 0;
-        RMT.rx_conf[channel].conf1.rx_en = 1;
-        RMT.rx_conf[channel].conf1.mem_owner = 1;
-        RMT.rx_conf[channel].conf1.mem_wr_rst = 1;
-    }
-#else
-    RMT.conf_ch[channel].conf0.div_cnt = 1;
-    RMT.conf_ch[channel].conf0.mem_size = buffers;
-    RMT.conf_ch[channel].conf0.carrier_en = 0;
-    RMT.conf_ch[channel].conf0.carrier_out_lv = 0;
-#if CONFIG_IDF_TARGET_ESP32
-    RMT.conf_ch[channel].conf0.mem_pd = 0;
-#endif
-    RMT.conf_ch[channel].conf0.idle_thres = 0x80;
-    RMT.conf_ch[channel].conf1.rx_en = 0;
-    RMT.conf_ch[channel].conf1.tx_conti_mode = 0;
-#if CONFIG_IDF_TARGET_ESP32
-    RMT.conf_ch[channel].conf1.ref_cnt_rst = 0;
-#else
-    RMT.conf_ch[channel].conf1.chk_rx_carrier_en = 0;
-#endif
-    RMT.conf_ch[channel].conf1.rx_filter_en = 0;
-    RMT.conf_ch[channel].conf1.rx_filter_thres = 0;
-    RMT.conf_ch[channel].conf1.idle_out_lv = 0;     // signal level for idle
-    RMT.conf_ch[channel].conf1.idle_out_en = 1;     // enable idle
-    RMT.conf_ch[channel].conf1.ref_always_on = 0;     // base clock
-
-    RMT.apb_conf.fifo_mask = 1;
+        rmt_config_t config = RMT_DEFAULT_ARD_CONFIG_RX(pin, channel, buffers);
+        ESP_ERROR_CHECK(rmt_config(&config));
+        ESP_ERROR_CHECK(rmt_driver_install(config.channel, 1024, 0)); //, rxBufferSize, 0));
```

Comment:
`rxBufferSize` is ignored here?

Reply:
Solved by keeping 1024 to all SoCs. Tested it with ESP32, S2 and C3. 
One possible solution is to use default values in the function signature, but that would require changing to change esp32-hal-rmt.c to cpp instead.